refactor(react): update useDocument to return `data` (#510)
sanity-io/sdk: packages/react/guides/0-Migration-Guide.md

- `UseProjectionOptions` → `useDocumentProjectionOptions`
- `UseProjectionResults` → `useDocumentProjectionResults`

3. Updated `useDocument` return structure

> All `useDocument` hooks now return their data under a `data` property for consistency with other hooks in the SDK.

Reviewer: ```suggestion
The `useDocument` hook now returns its data under a `data` property for consistency with other hooks in the SDK.
```

Changed to singular since there's only one `useDocument` hook ;)
